### PR TITLE
feat(json): use Ace Editor for JSON editing

### PR DESCRIPTION
**Pull request by anotherchrisberry**
I know how much @erikmunson likes the special JSON editor so I added it.

And fixed a couple of unrelated linting errors.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
     "@fortawesome/fontawesome-free": "5.5.0",
     "@types/plotly.js": "^1.28.12",
     "@uirouter/angularjs": "^1.0.15",
-    "@uirouter/react-hybrid": "^0.1.0",
+    "@uirouter/react-hybrid": "0.3.9",
```
Comment (anotherchrisberry) [inline package.json]: You can't actually load anything in the browser without this library update.

File: src/kayenta/canary.less [hunk 2]
```diff
@@ -43,8 +43,7 @@
     margin-bottom: 40px;
     overflow-y: auto;
     .config-detail-header {
-      padding-right: 15px;
-      padding-bottom: 7px;
+      padding: 5px 15px 7px 0;
```
Comment (anotherchrisberry) [inline src/kayenta/canary.less]: the "JSON" button retains focus after the modal closes, but there's no padding in the containing div so the border shadow only shows on 3 sides; adding some padding so it looks a little less bad

File: src/kayenta/edit/configJson.less [hunk 3]
```diff
@@ -1,3 +1,14 @@
+.ace-editor {
+  div,
+  span {
+    font-family: Menlo, Monaco, Consolas, 'Courier New', monospace;
+  }
+
+  > div.ace_gutter {
+    z-index: 0;
+  }
+}
```
Comment (erikmunson) [inline src/kayenta/edit/configJson.less]: do we have something that looks like this in `core`? or is this something specific to kayenta?

File: package.json [hunk 4]
```diff
@@ -55,7 +55,7 @@
     "ngimport": "^0.6.0",
     "plotly.js": "^1.35.2",
     "react": "16.3.2",
-    "react-ace": "6.1.4",
+    "react-ace": "6.4.0",
```
Comment (anotherchrisberry) [inline package.json]: effectively just a typings change to get the `debounceChangePeriod` prop

File: src/kayenta/edit/configJson.less (hunk 3 above)
Comment (anotherchrisberry) [reply src/kayenta/edit/configJson.less]: ooooh shoot I meant to do something about this, will do something about this, will be better than this

**Pull request merged**